core/state_processor.go

```diff
@@ -1102,35 +1138,36 @@ func ProcessQiTx(tx *types.Transaction, chain ChainContext, checkSig, isFirstQiT
 				ETXPCount++
 			}
 			if ETXRCount > *etxRLimit {
-				return nil, nil, fmt.Errorf("tx [%v] emits too many cross-region ETXs for block. emitted: %d, limit: %d", tx.Hash().Hex(), ETXRCount, etxRLimit), nil
+				return nil, 0, nil, nil, fmt.Errorf("tx [%v] emits too many cross-region ETXs for block. emitted: %d, limit: %d", tx.Hash().Hex(), ETXRCount, etxRLimit), nil
 			}
 			if ETXPCount > *etxPLimit {
-				return nil, nil, fmt.Errorf("tx [%v] emits too many cross-prime ETXs for block. emitted: %d, limit: %d", tx.Hash().Hex(), ETXPCount, etxPLimit), nil
+				return nil, 0, nil, nil, fmt.Errorf("tx [%v] emits too many cross-prime ETXs for block. emitted: %d, limit: %d", tx.Hash().Hex(), ETXPCount, etxPLimit), nil
 			}
 			primeTerminusHash := currentHeader.PrimeTerminusHash()
 			primeTerminusHeader := chain.GetHeaderByHash(primeTerminusHash)
 			if primeTerminusHeader == nil {
-				return nil, nil, fmt.Errorf("could not find prime terminus header %032x", primeTerminusHash), nil
+				return nil, 0, nil, nil, fmt.Errorf("could not find prime terminus header %032x", primeTerminusHash), nil
 			}
 			if !toAddr.IsInQiLedgerScope() {
-				return nil, nil, fmt.Errorf("tx [%v] emits UTXO with To address not in the Qi ledger scope", tx.Hash().Hex()), nil
+				return nil, 0, nil, nil, fmt.Errorf("tx [%v] emits UTXO with To address not in the Qi ledger scope", tx.Hash().Hex()), nil
 			}
 			if !chain.CheckIfEtxIsEligible(primeTerminusHeader.EtxEligibleSlices(), *toAddr.Location()) {
-				return nil, nil, fmt.Errorf("etx emitted by tx [%v] going to a slice that is not eligible to receive etx %v", tx.Hash().Hex(), *toAddr.Location()), nil
+				return nil, 0, nil, nil, fmt.Errorf("etx emitted by tx [%v] going to a slice that is not eligible to receive etx %v", tx.Hash().Hex(), *toAddr.Location()), nil
 			}
 
 			// We should require some kind of extra fee here
 			etxInner := types.ExternalTx{Value: big.NewInt(int64(txOut.Denomination)), To: &toAddr, Sender: common.ZeroAddress(location), OriginatingTxHash: tx.Hash(), ETXIndex: uint16(txOutIdx), Gas: params.TxGas}
```

Comment:
You didn't add an ETX type here

Reply:
If its not added, its the default type, still i should makje it explicit

Reply:
done